File: 18. Print nodes at level k.cpp
#/*
 Task: Print nodes at level k
 Time Complexity: O(n)
 Space Complexity: O(h) — recursion stack
*/
#include <bits/stdc++.h>
using namespace std;
struct Node
{
    int data;
    Node* left;
    Node* right;
    Node(int d) : data(d), left(nullptr), right(nullptr) {}
};
void printLevelK(Node* root, int k)
{
    if (!root)
        return;
    if (k == 0)
    {
        cout << root->data << " ";
        return;
    }
    printLevelK(root->left, k - 1);
    printLevelK(root->right, k - 1);
}
int main()
{
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);
    cout << "Nodes at level 2: ";
    printLevelK(root, 2);
    cout << endl;
    return 0;
}
// sc tc

File: 18. Print nodes at level k.py
class Node:
    def __init__(self,data): self.data=data; self.left=None; self.right=None

def print_level_k(root,k):
    if not root: return
    if k==0:
        print(root.data, end=' ')
        return
    print_level_k(root.left,k-1); print_level_k(root.right,k-1)

if __name__=='__main__':
    root=Node(1); root.left=Node(2); root.right=Node(3); root.left.left=Node(4); root.left.right=Node(5)
    print('Nodes at level 2:', end=' ')
    print_level_k(root,2); print()
